src/check_wheel_contents/checker.py

```diff
@@ -299,6 +299,29 @@ def check_W010(self, contents: WheelContents) -> list[FailedCheck]:
         else:
             return []
 
+    def check_W011(self, contents: WheelContents) -> list[FailedCheck]:
+        """
+        W011 — Wheel contains module and submodule with the same name
+
+        E.g: checks for both `path/foo.py` and `path/foo/__init__.py` being present
+        """
+        conflicts: list[str] = []
+
+        def check_dir(tree: Directory) -> None:
+            for name, entry in tree.entries.items():
+                if isinstance(entry, Directory):
+                    if tree.files.get(name + ".py") is not None:
+                        conflicts.append(entry.path or name)
```

Comment:
Instead of returning a single failed check that just contains the conflicting directory paths, I feel you should instead return one `FailedCheck` for each `foo.py` + `foo/__init__.py` pair, and the `FailedCheck` should contains both of those paths.

Cf. the following paragraph from ["Adding a New Check" in `CONTRIBUTING.md`](https://github.com/jwodder/check-wheel-contents/blob/fbedc168d570333a770167663d6fe56c556fff98/CONTRIBUTING.md#adding-a-new-check):

> Generally, a check method should return at most one `FailedCheck` containing all of the failing filepaths; only return multiple `FailedCheck`s if the check applies to groups of files.  For example, W002 ("Wheel contains duplicate files") returns a separate `FailedCheck` for each group of equal files.